em-client/src/client/mod.rs

```diff
@@ -6531,3 +6522,89 @@ impl error::Error for ClientInitError {
         "Failed to produce a hyper client."
     }
 }
+
+fn deserialize_config_checked(raw_config: &str, expected_hash: &[u8; 32]) -> Result<models::RuntimeAppConfig, ApiError> {
+    let result = serde_json::from_str::<models::RuntimeAppConfig>(&raw_config)
+        .map_err(|e| ApiError::new(format!("Failed to serialize RuntimeAppConfig to json: {}", e), SimpleErrorType::Permanent))?;
+
+    let hashed_config_part = serde_json::to_string(&result.config)
+        .map_err(|e| ApiError::new(format!("Failed to serialize HashedConfig to json: {}", e), SimpleErrorType::Permanent))?;
+
+    let hash = compute_sha256(hashed_config_part.as_bytes())
+        .map_err(|e| ApiError::new(format!("Unable to hash app config: {}", e), SimpleErrorType::Permanent))?;
+
+    if hash != *expected_hash {
+        Err(ApiError::new(format!("App config hash mismatch. Expected {:?}, but got {:?}. Raw App config is {}. Hashed part is {}", hash, expected_hash, raw_config, hashed_config_part).to_string(), SimpleErrorType::Permanent))
```

Comment:
It is enough to log only the hashes here. No need to log the app config itself, could be a data leak.

Reply:
Owww, that's a remainder from debugging stage